tensorflow/python/kernel_tests/decode_raw_op_test.py

```diff
@@ -57,5 +58,19 @@ def testToInt16(self):
           "size of int16"):
         decode.eval(feed_dict={in_bytes: ["123", "456"]})
 
+  def testToFloat16(self):
+    with self.test_session():
+      in_bytes = tf.placeholder(tf.string, shape=[None])
+      decode = tf.decode_raw(in_bytes, out_type=tf.float16)
+      self.assertEqual([None, None], decode.get_shape().as_list())
+
+      expected_result = np.matrix([[1, 2, 3, 4]], dtype=np.float16)
```

Comment:
I'm nitpicking here but you might want to consider feeding in a few negative values as well.